clients/cobol-lsp-vscode-extension/src/__tests__/services/LanguageClientServiceTest.spec.ts

```diff
@@ -102,17 +103,74 @@ describe("LanguageClientService positive scenario", () => {
   });
 
   test("Test LanguageClientService checkPrerequisites passes", async () => {
-    let message = false;
-    (fs.existsSync as any) = jest.fn().mockReturnValue(true);
-    vscode.workspace.getConfiguration(expect.any(String)).get = jest
-      .fn()
-      .mockReturnValue(9999);
+    let message: string | null = null;
+
+    const mockProcess = mockSpawnProcess(
+      "",
+      `java version "17.0.2" 2022-01-18 LTS
+      Java(TM) SE Runtime Environment (build 17.0.2+8-LTS-86)
+      Java HotSpot(TM) 64-Bit Server VM (build 17.0.2+8-LTS-86, mixed mode, sharing)`,
+      0,
+    );
+
     try {
       await languageClientService.checkPrerequisites();
-    } catch (error: any) {
-      message = error;
+    } catch (error: unknown) {
+      if (typeof error === "string") {
+        message = error;
+      }
     }
-    expect(message).toBeFalsy();
+    expect(message).toBeNull();
+    mockProcess.mockRestore();
```

Comment:
If the expect above fails. Does it still restore the mock?

Reply:
Yeah, I agree. It would be better to setup and restore mocks in before/after blocks, not it the test body. I'll fix that in some following PR.